File: alpha-terminal-ai-server/app/domains/youtube/domain/entity/youtube_video.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional


@dataclass
class YouTubeVideo:
    video_id: str
    title: str
    thumbnail_url: str
    channel_name: str
    published_at: str
    video_url: str
    view_count: int = 0
    id: Optional[int] = None
    created_at: datetime = field(default_factory=datetime.now)
    updated_at: datetime = field(default_factory=datetime.now)

File: frontend/features/youtube/domain/model/youtubeVideo.ts
export interface YoutubeVideo {
    title: string
    thumbnail_url: string
    channel_name: string
    published_at: string
    video_url: string
}

export interface YoutubeListResponse {
    items: YoutubeVideo[]
    next_page_token: string | null
    prev_page_token: string | null
    total_results: number
}
